genesyscloud/resource_genesyscloud_journey_outcome.go

```diff
@@ -235,12 +236,13 @@ func flattenJourneyOutcome(d *schema.ResourceData, journeyOutcome *platformclien
 	resourcedata.SetNillableValue(d, "is_positive", journeyOutcome.IsPositive)
 	resourcedata.SetNillableValue(d, "context", lists.FlattenAsList(journeyOutcome.Context, flattenContext))
 	resourcedata.SetNillableValue(d, "journey", lists.FlattenAsList(journeyOutcome.Journey, flattenJourney))
+
 	resourcedata.SetNillableValue(d, "associated_value_field", lists.FlattenAsList(journeyOutcome.AssociatedValueField, flattenAssociatedValueField))
 }
 
 func flattenAssociatedValueField(associatedValueField *platformclientv2.Associatedvaluefield) map[string]interface{} {
 	associatedValueFieldMap := make(map[string]interface{})
-	associatedValueFieldMap["dataType"] = associatedValueField.DataType
+	associatedValueFieldMap["data_type"] = associatedValueField.DataType
```

Comment:
This was another bug that only showed itself when the data condition was setup.  When an associated value field is set the schema value being set (dataType) is not defined in the schema.  It is data_type.  Terraform throws an exception if you try to set a schema value that is not defined in the schema.